### PR TITLE
.murdock.yml: add new Murdock configuration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR introduces some per repo configuration for the coming and under test [Murdock instance](https://ci-staging.riot-os.org).

The configurations provided should enable:
- comment on PRs with a link to the job and eventually some failing builds/tests, plus some stats
- skipping a job if the comment of the commit to build contains some keywords
- a downloadable result.json.gz file with a link on the Murdock UI of the job

### Testing procedure

- Green Murdock (both :) )
- Comments should be reported in this PR by riot-ci user
- Not sure but if some builds are triggered there should be a link to result.json.gz on the Murdock job page.

### Issues/PRs references

None

### CODE REVIEW

File: .murdock.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pr:
```
Comment (kaspar030) [inline .murdock.yml:1]: this will enable commenting, right? (should be disabled for now?)

Comment (kaspar030) [reply .murdock.yml:1]: ah, read description

**Comment by aabadie** [comment]
Hmm, the artifacts with a .json.gz file doesn't behave as I would expect. Will check how to improve that.

**Comment by kfessel** [comment]
Is this good to go? (or is further testing needed?)

**Comment by kfessel** [comment]
Is this good to go? (or is further testing needed?)

**Comment by aabadie** [comment]
> Is this good to go? (or is further testing needed?)

If we merge this now, all PRs will get a comment from the new Murdock instance with partial results (because jobs there are only performed on 5 targets). That might be confusing.

I'd rather wait that we switched to the new Murdock instance before we merge this one.

**Comment by aabadie** [comment]
> I'd rather wait that we switch to the new Murdock instance before we merge this one.

With the hope that the wait won't be too long.

**Comment by miri64** [comment]
> If we merge this now, all PRs will get a comment from the new Murdock instance with partial results (because jobs there are only performed on 5 targets). That might be confusing.

Can't the message maybe amended with "this is from the new Murdock testing instance, please confer to old Murdock for reliable results" to avoid such confusions? This way we can test the new Murdock to its fullest potential while it is still in staging.

**Comment by miri64** [comment]
That additional signature could be configured via `.murdock.yml` :-P This way, if we might need to add additional messages in the future ("Don't forget to register to the RIOT summit!" ^^) this will also be possible ;-).

**Comment by kaspar030** [comment]
> I'd rather wait that we switch to the new Murdock instance before we merge this one.

also, the single worker on murdock-ng is currently my workstation. that goes down when I need it elsewhere, e.g., for gaming. and I'd rather not have everyone get annoyed by CI timeout errors due to that.
We could use one of the other workers, but then production build times would go down...

**Comment by aabadie** [comment]
@miri64 your https://github.com/RIOT-OS/RIOT/pull/18084#issuecomment-1125207320 should now be addressed. We can now discuss what the footer should contain.

Note that the comment is posted twice because the PR is processed by 2 Murdock instances (on that will be switch to prod rather soonish and one that will be used as staging)

**Comment by aabadie** [comment]
> Note that the comment is posted twice because the PR is processed by 2 Murdock instances (on that will be switch to prod rather soonish and one that will be used as staging)

This is now fixed, only the prod instance can comment on pr, the feature is disabled globally on the staging instance.

**Comment by miri64** [comment]
> This is now fixed, only the prod instance can comment on pr, the feature is disabled globally on the staging instance.

Can we make this maybe a sticky comment (so a comment that changes instead of a new comment everytime)?

**Comment by miri64** [comment]
> @miri64 your [#18084 (comment)](https://github.com/RIOT-OS/RIOT/pull/18084#issuecomment-1125207320) should now be addressed. We can now discuss what the footer should contain.
> 

>> Don't forget to register to the [RIOT summit](https://summit.riot-os.org)!

Since there is no RIOT summit registration open yet, maybe not the best test line ;-). I guess for now, we can just leave it empty, since we went into production.

**Comment by aabadie** [comment]
> Can we make this maybe a sticky comment (so a comment that changes instead of a new comment everytime)?

Yes, I'll push an update to show how it behaves

> Since there is no RIOT summit registration open yet, maybe not the best test line ;-). I guess for now, we can just leave it empty, since we went into production.

Sure, I pushed that commit just show that it's possible. The footer contains your initial suggestions :)

**Comment by aabadie** [comment]
With sticky comments enabled, the first comment was edited (see https://github.com/RIOT-OS/RIOT/pull/18084#issuecomment-1122247013).  This might be confusing here, because the feature wasn't enabled initially.

**Comment by miri64** [comment]
> The footer contains your initial suggestions :)

I'll try to be more careful with my examples in the future :-)

**Comment by miri64** [comment]
Where did the success/failure table go btw?

**Comment by aabadie** [comment]
> Where did the success/failure table go btw?

There's none because the CI: skip compile test label is set, so no jobs were performed. I'll retrigger without.

**Comment by aabadie** [comment]
Hmm, the table didn't show up, will check later...

**Comment by aabadie** [comment]
> the table didn't show up, will check later...

Ok, the problem comes from the script that overwrites the job status (which contains failed/passed/total jobs stats) when building the documentation. This is only done for successful build jobs.

**Comment by aabadie** [comment]
The problem with the table is fixed now.

**Comment by kaspar030** [comment]
please squash!
ci-prod now has more than one worker and has been quite reliable lately.

**Review by kaspar030 — APPROVED** [review]
ACK.

**Pull request merged**